### PR TITLE
add builder provider

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

add BlobContainerClientBuilder as  we need new builder per request.

part of  fix: https://github.com/hmcts/blob-router-service/pull/378

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProvider.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.beans.factory.annotation.Lookup;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BlobContainerClientBuilderProvider {
+
+    @Lookup("bulk-scan-blob-client-builder")
+    public BlobContainerClientBuilder getBlobContainerClientBuilderBean() {
+        return null;
+    }
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProvider.java]: This is probably fine. Now that we have this provider, I wonder if the builder could just be created here. Thoughts?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProvider.java]: changed

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProvider.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import org.springframework.beans.factory.annotation.Lookup;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BlobContainerClientBuilderProvider {
+
+    @Lookup("bulk-scan-blob-client-builder")
+    public BlobContainerClientBuilder getBlobContainerClientBuilderBean() {
```
[the remark this answers is not in the document]
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProvider.java]: fixed

File: src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private BlobContainerClientBuilderProvider blobContainerClientBuilderProvider
+        = new BlobContainerClientBuilderProvider(httpClient, "http://example.com");
+
+    @Test
+    void should_provide_builder() {
+
+        BlobContainerClientBuilder blobContainerClientBuilder = blobContainerClientBuilderProvider
+            .getBlobContainerClientBuilder();
+
+        assertThat(blobContainerClientBuilder).isNotNull();
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProviderTest.java]: Is `httpClient` and `url` accessible from the client? Might be useful for assertions, if possible.

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientBuilderProviderTest.java]: not accessible

**Pull request merged**